lighthouse-core/test/lib/tracehouse/cpu-profile-model-test.js

```diff
@@ -167,31 +168,88 @@ describe('CPU Profiler Model', () => {
-        {ph: 'E', ts: ts(19.0e3), args: {data: {callFrame: {functionName: '(root)'}}}},
+        {ph: 'E', ts: ts(17.4e3), args: {data: {callFrame: {functionName: 'Baz'}}}},
+        {ph: 'E', ts: ts(17.4e3), args: {data: {callFrame: {functionName: 'Foo'}}}},
+        {ph: 'E', ts: ts(19.5e3), args: {data: {callFrame: {functionName: '(program)'}}}},
+        {ph: 'E', ts: ts(19.5e3), args: {data: {callFrame: {functionName: '(root)'}}}},
+      ]);
+    });
+
+    it('should handle multiple task start/stop times with low sampling rate', () => {
+      /*
+        An artistic rendering of the below profile with tasks:
+        ████████████████(root)████████████████
+        ███████Task███████  ██████Task██████ █
+         ██████Eval██████   ██████Eval██████
+         ██████Foo███████   ██████Bar██████
+          █Fn█
+          █Fn█
+      */
+      profile = {
+        id: '0x1',
+        pid: 1,
+        tid: 1,
+        startTime: 9e6,
+        nodes: [
+          {id: 0, callFrame: {functionName: '(root)'}},
+          {id: 1, callFrame: {functionName: 'Foo', url: 'fileA.js'}, parent: 0},
+          {id: 2, callFrame: {functionName: 'Bar', url: 'fileA.js'}, parent: 0},
+        ],
+        samples: [0, 1, 1, 2, 2, 0],
+        timeDeltas: [0.5e3, 19.5e3, 20e3, 20e3, 20e3, 20e3],
+      };
+
+      const ts = x => profile.startTime + x;
+
+      // With the sampling profiler we know that Foo switched to Bar, but we don't know when.
+      // Create a set of tasks that force large changes.
+      const tasks = [
+        // The RunTask at the toplevel, parent of Foo execution
+        {startTime: 1, endTime: 50, event: {ts: ts(1e3)}},
+        // The EvaluateScript at the next level, parent of Foo execution
+        {startTime: 5, endTime: 45},
+        // The FunctionCall at the next level, should be a child of Foo execution
+        {startTime: 10, endTime: 25},
+        // The FunctionCall at the next level, should be a child of Foo execution
+        {startTime: 12, endTime: 22},
+        // The RunTask at the toplevel, parent of Bar execution
+        {startTime: 51, endTime: 90},
+        // The EvaluateScript at the next level, parent of Bar execution
+        {startTime: 55, endTime: 85},
+        // The RunTask at the toplevel, there to mess with Bar timing
+        {startTime: 92, endTime: 103},
+      ];
+
+      const events = CpuProfilerModel.synthesizeTraceEvents(profile, tasks);
+
+      expect(events).toMatchObject([
+        {ph: 'B', ts: ts(0.5e3), args: {data: {callFrame: {functionName: '(root)'}}}},
+        {ph: 'B', ts: ts(5e3), args: {data: {callFrame: {functionName: 'Foo'}}}},
+        {ph: 'E', ts: ts(45e3), args: {data: {callFrame: {functionName: 'Foo'}}}},
+        {ph: 'B', ts: ts(55e3), args: {data: {callFrame: {functionName: 'Bar'}}}},
+        {ph: 'E', ts: ts(85e3), args: {data: {callFrame: {functionName: 'Bar'}}}},
+        {ph: 'E', ts: ts(103e3), args: {data: {callFrame: {functionName: '(root)'}}}},
```

Comment:
(root) from the cpu sampler takes the full duration.

but then the tasks above indicate we had 3 toplevel RunTasks.  

how does that work?

Reply:
From some real traces that's how it appears to work in practice with `(idle)` filling in the time in between the toplevel tasks. I can add another test for multiple roots though 👍

Reply:
yeah it seems like (root), (idle) and (program) are are basically unlike the rest of the nodes in a cpu profile, so it feels like they dont belong as trace events.

Reply:
yeah the integration PR I have where I have been finding the other failure modes filters those out :)